src/acme/ui/CustomerListUI.java

```diff
@@ -0,0 +1,319 @@
+package acme.ui;
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Font;
+import java.awt.Panel;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.sql.ResultSet;
+
+import javax.swing.DefaultCellEditor;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JTextField;
+import javax.swing.SwingConstants;
+import javax.swing.table.TableCellEditor;
+import javax.swing.table.TableCellRenderer;
+import javax.swing.table.TableColumn;
+
+import acme.pd.User;
+import acme.pd.Customer;
+import acme.pd.MapIntersection;
+import acme.pd.Road;
+
+public class CustomerListUI extends AcmeBaseJPanel {
+  
+  public CustomerListUI() {
+    super();
+  }
+  
+  public void buildPanel() {
+  
+    setLayout(null);
+    
+    //title label
+    JLabel lblCustomers = new JLabel("Customers");
+    lblCustomers.setFont(new Font("Lucida Grande", Font.PLAIN, 24));
+    lblCustomers.setBounds(20, 20, 200, 27);
+    add(lblCustomers);
+    
+    //my test data
+    Customer[] testData = getTestData();
+    
+
+    //scroll and table. I made this in netbeans, so it looks a little different
+    JScrollPane jScrollPane1 = new JScrollPane();
+    JTable jTable1 = new JTable();
+
+    //set up table
+    jTable1.setBorder(javax.swing.BorderFactory.createEtchedBorder());
+    jTable1.setFont(new java.awt.Font("Lucida Grande", 0, 16)); 
+    jTable1.setModel(new javax.swing.table.DefaultTableModel(
+        new Object [][] {
```

Comment:
Will this work in the future with an arbitrary length customer list or array?

Reply:
There would have to be some reworking, but I think I could get it there.